src/sinks/util/rusoto.rs

```diff
@@ -63,7 +63,14 @@ impl AwsCredentialsProvider {
             let creds = AutoRefreshingProvider::new(provider).context(InvalidAWSCredentials)?;
             Ok(Self::Role(creds))
         } else {
-            let creds = DefaultCredentialsProvider::new().context(InvalidAWSCredentials)?;
+            debug!("using default credentials provider for AWS.");
+            let mut chain = ChainProvider::new();
+            // 8 seconds because our default healthcheck timeout
```

Comment:
Can we not coordinate these timeouts through a shared variable or configuration setting?

Reply:
I don't think that would make much sense, this is the only thing I know of that actually comes with its own timeout built in. We generally want to lean on the timeout in topology but in this case the error was not surfacing fast enough.

Reply:
What if we decide to reduce our topology timeout to 5 seconds, for example?

Reply:
The proper solution here is to rewrite this credential fetcher to not use a timeout. I'll open an issue for that.

Reply:
It's a network request, right? So it in inherently has a timeout of some sort.

Reply:
I opened https://github.com/timberio/vector/issues/2344 which should provide a better path to fixing these issues.

Reply:
> It's a network request, right? So it inherently has a timeout of some sort.

There are network requests involved but we should be apply the timeout at the root but I also think we should be bailing out quickly instead of looping and retrying. I think we just want more control over the process so we can provide the right info. That for example we started a instance metadata request but it never finished.

Reply:
I think it's reasonable to retry up to limit unless these network requests should never fail, which doesn't sound like a network request.

Reply:
Yeah, I think the goal with that issue is to dig into this more and see if we can write a better version that gives more visibility.